### DISCUSSION TITLE
Dynamic and Iterable providers - thank you

### DISCUSSION THREAD

**Discussion by ghost** [opentofu/opentofu#2125] [General]
I've been waiting a very long time to use Terraform/OpenTofu due to the limitations of providers and their inability to be dynamic, iterable, or generated at runtime. It always seemed like such an oversight in the original design, or an over-protection of end users in order to avoid providing footguns. Having this basic functionality will allow proper DRY principles and simplify tofu code across the board -in addition more powerful generalized modules.

Thank you for putting the community needs in favor of arbitrary decisions, and thank you for the hard work on this project!

**Comment by cam72cam** [comment]
Hope you are enjoying the 1.9 Release Candidate (and soon release)!

**Comment by ghost** [comment]
Very much so, I was able to get rid of many bash/ps scripts that were being used for scaffolding and simplify some pipelines. I can finally start to migrate my more customized provisioning over to Tofu as well due to another thing that was added `-excludes` flag. 

Best release so far hands down. Thank you!
